app/src/main/java/com/firebase/uidemo/database/ChatIndexActivity.java

```diff
@@ -34,12 +35,13 @@ public void onClick(View v) {
 
     @Override
     protected FirebaseRecyclerAdapter<Chat, ChatHolder> getAdapter() {
-        return new FirebaseIndexRecyclerAdapter<Chat, ChatHolder>(
-                Chat.class,
+        return new FirebaseRecyclerAdapter<Chat, ChatHolder>(
+                new FirebaseIndexArray<>(
```

Comment:
Yeah, I wasn't sure about that. I only added the `LifecycleOwner` parameter to the constructor that takes an `ObservableSnapshotArray` because it looked insane to have 7 constructor overloads.

If you think we should provide an overload with a `LifecycleOwner` for each constructor then I'll be happy to add them, I just wasn't sure if we should go that route.